### PR TITLE
tests for non-multiprocessing management commands

### PR DESCRIPTION
**Pull request by ajslater**
Add tests for xapian haystack management commands.
Does not include a multiprocessing management command test with multiple workers as that is being addressed in https://github.com/notanumber/xapian-haystack/pull/219

### CODE REVIEW

File: tests/xapian_tests/tests/test_management_commands.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        call_command("rebuild_index", interactive=False, verbosity=0)
+        self.verify_indexed_documents()
+
+    def test_remove(self):
+        call_command("clear_index", interactive=False, verbosity=0)
+        self.verify_indexed_document_count(0)
+
+        call_command("update_index", verbosity=0)
+        self.verify_indexed_documents()
```
Comment (claudep) [inline tests/xapian_tests/tests/test_management_commands.py:70]: I don't see the utility of clear_index/update_index sequence here.

File: tests/xapian_tests/tests/test_management_commands.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return BlogSearchIndex()
+
+    def setUp(self):
+        super().setUp()
+
+        self.sample_objs = []
+
+        for i in range(1, self.NUM_BLOG_ENTRIES + 1):
+            entry = BackendFeaturesTestCase.get_entry(i)
+            entry.float_number = random.uniform(0.0, 1000.0)
```
Comment (claudep) [inline tests/xapian_tests/tests/test_management_commands.py:25]: What's the intent with rewriting the float value?

File: tests/xapian_tests/tests/test_management_commands.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    def test_remove(self):
+        call_command("clear_index", interactive=False, verbosity=0)
+        self.verify_indexed_document_count(0)
+
+        call_command("update_index", verbosity=0)
+        self.verify_indexed_documents()
+
+        # Remove several instances.
+        BlogEntry.objects.get(pk=1).delete()
```
Comment (claudep) [inline tests/xapian_tests/tests/test_management_commands.py:73]: I would avoid counting on a hardcoded pk, `BlogEntry.objects.all()[0]` should work. Same below.

**Review by claudep — CHANGES_REQUESTED** [review]
Great, thanks, some minor questions remain.

File: tests/xapian_tests/tests/test_management_commands.py (hunk 1 above)
Comment (ajslater) [reply tests/xapian_tests/tests/test_management_commands.py:71]: It appears the whoosh tests use call_command() to do manual set setup, which isn't ideal. This has been replaced in the latest revision.

File: tests/xapian_tests/tests/test_management_commands.py (hunk 2 above)
Comment (ajslater) [reply tests/xapian_tests/tests/test_management_commands.py]: The tests from which these were copied gave the entries unique values, but these tests do not make use of those values, so it's been removed.

File: tests/xapian_tests/tests/test_management_commands.py (hunk 3 above)
Comment (ajslater) [reply tests/xapian_tests/tests/test_management_commands.py]: This has been changed not to rely on hardcoded pk values

**Comment by ajslater** [comment]
These tests were cut and pasted from the django-haystack whoosh tests. You've correctly identified that those tests are not written to precisely isolate the commands they are testing. This latest revision should be a little better. It has separate tests for four different management commands rather than lumping them together and does not use management commands to set up the tests.

**Pull request merged**